index.js

```diff
@@ -401,6 +402,49 @@ function pushAvailableLocation(ctx, locationsAvailable, locationIdx) {
   }
 }
 
+/** Helper function to see if a set is strictly a subset of another. */
+function isSubset(firstSet, secondSet) {
+  let allPresent = true
+
+  firstSet.forEach(function(element) {
+    allPresent = allPresent && secondSet.has(element)
+  })
+
+  return allPresent
+}
+
+/**
+ * Helper function to take a list of locks and replace all instances
+ * of a single ability with new locks
+ *
+ * @param {Array} locks - A list of locks to update
+ * @param {String} abilityToReplace - The ability to replace requirements of
+ * @param {Array} locksToAdd - A list of lock sets to replace the original
+ *   ability requirement with
+ */
+function replaceLocks(locks, abilityToReplace, locksToAdd) {
+  const newLocks = []
+  locks.forEach(function(lock) {
+    if (!lock.has(abilityToReplace)) {
+      // Any locks that didn't require the relic can stay the same
+      newLocks.push(lock)
+    }
+    else {
```

Comment:
Style nit: put the `else` on the same line as the `if` statement's closing curly paren.